src/components/icons/ThemedIcons.tsx

```diff
@@ -27,56 +26,79 @@ export interface IconProps {
 export type IconComponent = React.FunctionComponent<IconProps>
 
 //
-// HOCs
+// Inner components
 //
 
-function makeAnimatedFontIcon(IconComponent: React.ComponentType<AnimatedProps<VectorIconProps>>, name: string): AnimatedIconComponent {
-  return (props: AnimatedIconProps) => {
-    const { accessible, color, size } = props
-    const { icon, rem } = useTheme()
-    const oneRem = rem(1)
+interface IconChoice {
+  IconComponent: typeof Icon
+  name: string
+}
+
+function AnimatedFontIcon(props: AnimatedIconProps & IconChoice): JSX.Element {
+  const { accessible, color, IconComponent, name, size } = props
+  const theme = useTheme()
+  const defaultColor = theme.icon
+  const defaultSize = theme.rem(1)
+
+  const fontFamily = IconComponent.getFontFamily()
+  const glyphMap = IconComponent.getRawGlyphMap()
+
+  const style = useAnimatedStyle(() => ({
+    color: color?.value ?? defaultColor,
+    fontFamily,
+    fontSize: size?.value ?? defaultSize,
+    fontStyle: 'normal',
+    fontWeight: 'normal'
+  }))
+
+  return (
+    <Animated.Text accessible={accessible} style={style}>
+      {String.fromCodePoint(glyphMap[name])}
+    </Animated.Text>
+  )
```

Comment:
We should document this approach with comments. Something to mention is the importance of using a single Animated.Text component for animated icons to avoid the buggy behavior in reanimated. Also, mention the limitation that switching between animated icons should be avoided completely in favor of parameterizing the icon name prop instead; this may limit our ability to switch between icons in different typesets.